attestation-agent/attester/src/nvidia/mod.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2025 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::{Attester, TeeEvidence};
+use anyhow::{bail, Context, Result};
+use base64::{engine::general_purpose::STANDARD, Engine};
+use log::warn;
+use nvml_wrapper::{enums::device::DeviceArchitecture, Nvml};
+use serde::Serialize;
+
+pub fn detect_platform() -> bool {
+    // Return true iff one GPU is found and it has CC mode set.
+    match Nvml::init() {
+        Ok(nvml) => {
+            nvml.device_count().is_ok_and(|count| count == 1)
+                && nvml
+                    .device_by_index(0)
+                    .is_ok_and(|device| device.is_cc_enabled().unwrap_or_default())
+        }
+        Err(_) => false,
+    }
+}
+
+/// NRAS knows about "switch" and "gpu" but the expected evidence
+/// content is the same. nvidia-attester can compose a list of
+/// all CC enabled nvml/nscq devices using this evidence struct.
+#[derive(Serialize)]
+struct NvDeviceReportAndCert {
+    arch: DeviceArchitecture,
+    uuid: String,
+    evidence: String,
+    certificate: String,
+}
+
+#[derive(Serialize)]
+struct NvDeviceEvidence {
+    device_evidence_list: Vec<NvDeviceReportAndCert>,
+}
+
+#[derive(Debug, Default)]
+pub struct NvAttester {}
+
+#[async_trait::async_trait]
+impl Attester for NvAttester {
+    async fn get_evidence(&self, report_data: Vec<u8>) -> Result<TeeEvidence> {
+        let nvml = Nvml::init()?;
+        let devices = nvml.device_count()?;
+
+        let mut device_evidence_list = vec![];
+
+        if report_data.len() < 32 {
+            warn!("NVIDIA Attester: Report data must be at least 32 bytes");
+        }
+
+        let nonce: [u8; 32] = report_data[0..32].try_into()?;
```

Comment:
note that if it is longer than 32 bytes, it will be truncated. also, if it is less than 32 bytes, the indexing `[]0..32]` here will panic. We can use soome similar logic as https://github.com/confidential-containers/guest-components/blob/main/attestation-agent/attester/src/tdx/mod.rs#L127

Reply:
updated